packages/fuel-indexer-plugin/src/wasm.rs

```diff
@@ -128,22 +128,39 @@ pub trait Entity<'a>: Sized + PartialEq + Eq + std::fmt::Debug {
     }
 
     /// Finds the first entity that satisfies the given constraints.
-    fn find(query: impl Into<QueryFragment<Self>>) -> Option<Self> {
-        let query: QueryFragment<Self> = query.into();
+    fn find(query: impl ToFilter<Self>) -> Option<Self> {
+        let result = Self::find_many(1, query);
+        result.into_iter().next()
+    }
+
+    /// Finds the entities that satisfy the given constraints.
+    fn find_many(limit: usize, query: impl ToFilter<Self>) -> Vec<Self> {
```

Comment:
- So the semantics of this `find_many` are a bit off.
- The `LIMIT` clause in SQL works similar to the `ORDER BY` clause
- In the `find` implementation https://github.com/FuelLabs/fuel-indexer/pull/1446, we added the `order_by` clause as a separate method

```rust
Entity::find(Entity::field().eq(value).order_by(Entity::field()).asc())
```
- The `LIMIT` clause works in a similar way - as a separate method
```rust
Entity::find(Entity::field().eq(value).order_by(Entity::field()).asc().limit(5))
```
- With that said, `.limit()` is a separate piece of functionality from `.find_many`
  - Totally fine if you want to go ahead and throw both of those in the same PR
  - When I mentioned "limit the result set" in the original issue, I was more so making sure we're aware of how much space we're allocating in the runtime for the result set

References
- [Sequelize `limit`](https://sequelize.org/docs/v6/core-concepts/model-querying-basics/#limits-and-pagination)
- [Django](https://docs.djangoproject.com/en/dev/topics/db/queries/#limiting-querysets) is actually notoriously known for not having a `.limit()` API 😆 
- [Diesel `.limit()`](https://docs.diesel.rs/2.0.x/diesel/prelude/trait.QueryDsl.html#method.limit)